Classes/SoundFileAnalysis.sc

```diff
@@ -49,18 +49,33 @@ SoundFileAnalysis {
 		^result
 	}
 
-	analyzeFile { |path, start = (0), duration, which, callback, maxDataPoints = (1000)|
+	analyzeFile { |path, start = (0), duration, which, callback, maxDataPoints = (1000),trimDuration=\roundDown, serverOptions|
 		var result = ();
 		if(File.exists(path).not) { "\nFile not found: %\n".format(path).warn; ^this };
 		which = (which ?? { analysisMethods.keys.as(Array).sort }).asArray;
 		fork {
 			var resultpaths, oscpath, score;
-			var analysisDuration, soundFile, cond;
+			var analysisDuration, soundFile, options, cond;
 			var server = Server(\dummy);
 
 			// get duration and numChannels from soundFile
 			soundFile = SoundFile.openRead(path);
-			analysisDuration = min(duration ?? { soundFile.duration - start }, soundFile.duration);
+
+			options= (serverOptions ?? {ServerOptions()})
+				.verbosity_(-1)
+				.memSize_(8192 * 256) // REALLY NEEDED?
+				.sampleRate_(soundFile.sampleRate);
+			);
```

Comment:
Here is something I wonder about: if you pass in server options, wouldn't you be surprised that they are overridden? For example, we may want to oversample the analysis and use a higher sample rate than the sound file, etc.

Perhaps it's better to provide a class variable with `defaultServerOptions`? … and then do something like this:
```supercollider
options = serverOptions ? defaultServerOptions;
if(options.sampleRate.isNil) { options = options.copy.sampleRate_(soundFile.sampleRate) };
```